### PR TITLE
feat(DLI): add a new resource to manage authentication

### LINKED ISSUE #1670: [DLI]Need a new resource to manage Datasource Authentication
### Current Terraform Version

```
...
```

### Proposal

1. user create an Authentication Certificate

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1670 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccDatasourceAuth'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccDatasourceAuth -timeout 360m -parallel 4
=== RUN   TestAccDatasourceAuth_basic
=== PAUSE TestAccDatasourceAuth_basic
=== RUN   TestAccDatasourceAuth_css
=== PAUSE TestAccDatasourceAuth_css
=== RUN   TestAccDatasourceAuth_Kafka_SSL
=== PAUSE TestAccDatasourceAuth_Kafka_SSL
=== RUN   TestAccDatasourceAuth_KRB
=== PAUSE TestAccDatasourceAuth_KRB
=== CONT  TestAccDatasourceAuth_basic
=== CONT  TestAccDatasourceAuth_Kafka_SSL
=== CONT  TestAccDatasourceAuth_css
=== CONT  TestAccDatasourceAuth_KRB
--- PASS: TestAccDatasourceAuth_KRB (10.42s)
--- PASS: TestAccDatasourceAuth_basic (13.54s)
--- PASS: TestAccDatasourceAuth_css (16.21s)
--- PASS: TestAccDatasourceAuth_Kafka_SSL (18.70s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       18.738s
```

### CODE REVIEW

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+				Computed:    true,
+				Description: `Username.`,
+				ConflictsWith: []string{
+					"truststore_location",
+				},
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go:71]: **Sensitive** is missing

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go [hunk 2]
```diff
@@ -144,24 +144,27 @@ func ResourceCssCluster() *schema.Resource {
 			},
 
 			"vpc_id": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-				Computed: true,
+				Type:        schema.TypeString,
```
Comment (ShiChangkuo) [inline huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: please delete this file

File: docs/resources/dli_datasource_auth.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    + **passwd**: Password.
+    + **CSS**: CSS.
+    + **KRB**: Kafka_SSL.
+    + **Kafka_SSL**: Kafka_SSL.
+
+  Changing this parameter will create a new resource.
+
+* `username` - (Optional, String) Username.
+
+* `password` - (Optional, String) Password.
```
Comment (ShiChangkuo) [inline docs/resources/dli_datasource_auth.md]: please add more information for the two parameters.

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+
+	getDatasourceAuthRespBody, err := utils.FlattenResponse(getDatasourceAuthResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("auth_infos[0].auth_info_name", getDatasourceAuthRespBody, nil)),
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go:248]: I think you can save the "auth_infos[0]", and check whether it exist

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+		"truststore_location",
+		"truststore_password",
+		"keystore_location",
+		"keystore_password",
+		"krb5_conf",
+		"keytab",
+	}
+
+	if d.HasChanges(updateDatasourceAuthHostsChanges...) {
+		// updateDatasourceAuthHosts: update hosts
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go]: the comment is wrong

File: docs/resources/dli_datasource_auth.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+```
+
+Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
+API response, security or some other reason. The missing attributes include:
+`password`, `truststore_password`, `keystore_password`, `key_password`.
+It is generally recommended running `terraform plan` after importing a resource.
+You can then decide if changes should be applied to the resource, or the resource definition should be updated to align
+with the resource. Also you can ignore changes as below.
+
+```bash
```
Comment (Jason-Zhang9309) [inline docs/resources/dli_datasource_auth.md]: hcl

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go:71]: done

File: docs/resources/dli_datasource_auth.md (hunk 3 above)
Comment (hmbig2) [reply docs/resources/dli_datasource_auth.md]: done

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go:248]: ok

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go (hunk 5 above)
Comment (hmbig2) [reply huaweicloud/services/dli/resource_huaweicloud_dli_datasource_auth.go]: fixed

File: docs/resources/dli_datasource_auth.md (hunk 6 above)
Comment (hmbig2) [reply docs/resources/dli_datasource_auth.md]: done

**Pull request merged**